### PR TITLE
Ryna - L2052.ag_prodchange_cost_input/irr/mgmt 

### PR DESCRIPTION
**Pull request by rynacui**
This is a triplet of `L205`, `L2051`, and `L2052`.  Have not created the xml generation chunk.

### CODE REVIEW

File: R/zchunk_L2052.ag_prodchange_cost_irr_mgmt.R [hunk 1]
```diff
@@ -1,172 +1,319 @@
...
 #'
-#' Briefly describe what this chunk does.
+#' Specifies production costs and future agricultural productivity changes for all technologies.
 #'
 #' @param command API command to execute
 #' @param ... other optional parameters, depending on command
 #' @return Depends on \code{command}: either a vector of required inputs,
 #' a vector of output names, or (if \code{command} is "MAKE") all
-#' the generated outputs: \code{L2052.AgCost_ag_irr_mgmt}, \code{L2052.AgCost_bio_irr_mgmt}, \code{L2052.AgCost_For}, \code{L2052.AgProdChange_ag_irr_ref}, \code{L2052.AgProdChange_bio_irr_ref}, \code{L2052.AgProdChange_irr_high}, \code{L2052.AgProdChange_ag_irr_ref}, \code{L2052.AgProdChange_irr_low}, \code{L2052.AgProdChange_irr_ssp4}, \code{L2052.AgProdChange_irr_high}. The corresponding file in the
+#' the generated outputs: \code{L2052.AgCost_ag_irr_mgmt}, \code{L2052.AgCost_bio_irr_mgmt}, \code{L2052.AgCost_For}, \code{L2052.AgProdChange_ag_irr_ref}, \code{L2052.AgProdChange_bio_irr_ref}, \code{L2052.AgProdChange_irr_high}, \code{L2052.AgProdChange_irr_low}, \code{L2052.AgProdChange_irr_ssp4}. The corresponding file in the
```
Comment (rynacui) [inline R/zchunk_L2052.ag_prodchange_cost_irr_mgmt.R]: Delete repeated outputs

File: R/zchunk_L2052.ag_prodchange_cost_irr_mgmt.R [hunk 2]
```diff
@@ -1,172 +1,319 @@
...
+             FILE = "temp-data-inject/L201.AgYield_bio_tree",
+             "L102.pcgdp_thous90USD_Scen_R_Y"))
   } else if(command == driver.DECLARE_OUTPUTS) {
     return(c("L2052.AgCost_ag_irr_mgmt",
-"L2052.AgCost_bio_irr_mgmt",
-"L2052.AgCost_For",
-"L2052.AgProdChange_ag_irr_ref",
-"L2052.AgProdChange_bio_irr_ref",
-"L2052.AgProdChange_irr_high",
-"L2052.AgProdChange_ag_irr_ref",
```
Comment (rynacui) [inline R/zchunk_L2052.ag_prodchange_cost_irr_mgmt.R]: Repeated outputs, deleted.

File: R/zchunk_L2052.ag_prodchange_cost_irr_mgmt.R [hunk 3]
```diff
@@ -1,172 +1,319 @@
...
+
+    L2052.AgProdChange_irr_high %>%
+      add_title("High agricultural productivity change of crops by region / crop / GLU / technology") %>%
+      add_units("Unitless") %>%
+      add_comments("Multiply reference productivity change with the high growth rate multiplier") %>%
+      add_comments("The same productivity change are assigned to both high and low management") %>%
+      add_legacy_name("L2052.AgProdChange_irr_high") %>%
+      same_precursors_as("L2052.AgProdChange_ag_irr_ref") %>%
+      add_precursors("L102.pcgdp_thous90USD_Scen_R_Y") %>%
+      add_flags(FLAG_PROTECT_FLOAT) ->
```
Comment (rynacui) [inline R/zchunk_L2052.ag_prodchange_cost_irr_mgmt.R]: Need to protect float to pass the test.

File: R/zchunk_L2052.ag_prodchange_cost_irr_mgmt.R [hunk 4]
```diff
@@ -1,172 +1,319 @@
...
+      add_title("SSP4 agricultural productivity change of crops by region / crop / GLU / technology") %>%
+      add_units("Unitless") %>%
+      add_comments("Assign reference productivity change to median income regions") %>%
+      add_comments("Assign high productivity change to high income regions") %>%
+      add_comments("Assign low productivity change to low income regions") %>%
+      add_comments("Region groups by income level are based on the 2010 GDP per capita") %>%
+      add_legacy_name("L2052.AgProdChange_irr_ssp4") %>%
+      same_precursors_as("L2052.AgProdChange_ag_irr_ref") %>%
+      add_precursors("L102.pcgdp_thous90USD_Scen_R_Y") %>%
+      add_flags(FLAG_PROTECT_FLOAT) ->
```
Comment (rynacui) [inline R/zchunk_L2052.ag_prodchange_cost_irr_mgmt.R]: Need to protect float to pass the test

File: R/zchunk_L2052.ag_prodchange_cost_irr_mgmt.R [hunk 5]
```diff
@@ -1,172 +1,319 @@
...
+    L201.AgYield_bio_tree <- get_data(all_data, "temp-data-inject/L201.AgYield_bio_tree")
+    L102.pcgdp_thous90USD_Scen_R_Y <- get_data(all_data, "L102.pcgdp_thous90USD_Scen_R_Y")
+
+    # Define column names
+    names_AgTech <- LEVEL2_DATA_NAMES[["AgTech"]]
+    names_AgCost <- LEVEL2_DATA_NAMES[["AgCost"]]
+    names_AgProdChange <- LEVEL2_DATA_NAMES[["AgProdChange"]]
+
+    # Production costs
+    # Assign nonLandVariableCost of crop prodcution, assuming the same level to all four technologies
```
Comment (kvcalvin) [inline R/zchunk_L2052.ag_prodchange_cost_irr_mgmt.R]: `production`

**Comment by rynacui** [comment]
Thanks @bpbond! I'll then go back to my pending demand chunk.

**Pull request merged**